ext/rotoscope/strmemo.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _INC_ROTOSCOPE_STRMEMO_H_
+#define _INC_ROTOSCOPE_STRMEMO_H_
+
+#include "uthash.h"
+
+typedef struct {
+  unsigned long key;
+  UT_hash_handle hh;
+} rs_strmemo_t;
+
+bool rs_strmemo_uniq(rs_strmemo_t **calls, unsigned char *entry);
+void rs_strmemo_free(rs_strmemo_t *calls);
```

Comment:
Sad that Ruby doesn't expose its own hash for use in C extensions (although that would probably be slower because of constructing Ruby objects...)